packages/react-juce/src/components/Button.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+
+import { View } from './View'
+import { SyntheticMouseEvent } from '../lib/SyntheticEvents'
+import { ViewInstance } from "../lib/BlueprintBackend";
+
+//TODO: Once ViewProps work is complete we can probably
+//      remove this in favour of ViewProps.
+export interface ButtonProps {
+  onClick: (e: SyntheticMouseEvent) => void;
+  onMouseDown?: (e: SyntheticMouseEvent) => void;
+  onMouseUp?: (e: SyntheticMouseEvent) => void;
+}
+
+type ButtonState = {
+  down: boolean
+}
+
+/**
+ * A simple Button component which can be used as a building block
+ * for more complex button types.
+ *
+ * @example
+ *
+ * <Button onClick={() => { console.log("clicked"); }} {...styles.button}>
+ *   <Text {...styles.text}>
+       Hello World
+ *   </Text>
+ * </Button>
+ *
+ * const styles = {
+ *   text: {
+ *     'font-size': 18.0,
+ *     'line-spacing': 1.6,
+ *     'color': 'ff626262'
+ *   },
+ *   button: {
+ *     'justify-content': 'center',
+ *     'align-items': 'center',
+ *     'width': '100%',
+ *     'height': '100%',
+ *     'border-radius': 5.0,
+ *     'border-width': 2.0,
+ *     'border-color': 'ff626262'
+ *   }
+ * };
+ *
+ */
+export class Button extends Component<ButtonProps, ButtonState> {
+  private _ref: React.RefObject<ViewInstance>;
+
+  constructor(props: ButtonProps) {
+    super(props);
+    this._ref = React.createRef();
+
+    this.state = {
+      down: false
+    }
+  }
+
+  handleDown = (e: SyntheticMouseEvent): void => {
+    if (this.props.onMouseDown && typeof this.props.onMouseDown === 'function')
```

Comment:
This can just be `if (typeof this.props.onMouseDown === 'function')`. If `this.props.onMouseDown` does not exist, then `typeof this.props.onMouseDown` will not === 'function'

Reply:
Cool. Might have done that a couple times actually so will clean up.